### DISCUSSION TITLE
How to create Custom Handlers (wiki documentation incorrect)

### DISCUSSION THREAD

**Discussion by headintheclouds21** [dotnet/maui#20929] [General]
I've been recently trying to create custom handlers in NET MAUI and I came across this page on the wiki. https://github.com/dotnet/maui/wiki/Porting-Custom-Renderers-to-Handlers#net-maui-handlers I've tried implmenting this, but was unsuccessful. I've noticed that some of the code appears to be incorrect and prevents the example from building. 

For example, this line in the wiki, 'collection' should be replaced with 'handlers'
`builder.ConfigureMauiHandlers(collection =>`
        `{`
`//#if __ANDROID__
`
           ` handlers.AddHandler(typeof(CustomEntry), typeof(CustomEntryHandler));`
`//#endif`
 ` });`
 
   This leads me to believe there are other syntax issues too could be preventing it from working as expected, Does anyone know the correct way to create the custom handler in the example. The example is for an Android EditText. I'd like to create one that control as well as other android controls. If i can get a working example to help me understand how this works, I can build off that knowledge. Seeing a complete example would be very helpful. I've already read through the MAUI docs. Any advice is greatly appreciated

**Comment by davidortinau** [comment]
Have you reviewed the Learn documentation on this? That page is old. I'll make sure it links to these docs.

https://learn.microsoft.com/en-us/dotnet/maui/migration/renderer-to-handler?view=net-maui-8.0
https://learn.microsoft.com/en-us/dotnet/maui/user-interface/handlers/create?view=net-maui-8.0

I have a few handlers here that might be a good reference https://github.com/davidortinau/SmartHotel360-Mobile/tree/main/Source/SmartHotel.Clients/SmartHotel.Clients

**Reply by headintheclouds21** [reply]
Hi @davidortinau thanks for your reply. Those links were helpful, but I'm still struggling to fully grasp how to create custom handlers for android specific controls. The example on the older docs was creating one for the Android EditText control. That's one of the controls I'm trying to create a handler for in my own app. Seeing an example of this should look would significantly help me. I wish those docs were up to date. Is there any chance you could help me create a handler for Android EditText or show me what it should look like? A working example of that would go a long ways.

**Reply by pictos** [reply]
@headintheclouds21 I created this [playlist with videos](https://www.youtube.com/playlist?list=PL3GeZ8t0AGwwlDKbbDOjDNbbeNUF4lwvj) on how to customize controls using handlers, the videos are in portuguese, but youtube should be able to create translated captions for it. From what you're saying, what you need is to use the `ViewHandler.PlatformViewFactory`, [here's a blog post](https://blog.jhonatanoliveira.dev/customize-controls-with-net-maui-using-platformviewfactory) that may help you

**Reply by headintheclouds21** [reply]
Hi @pictos I appreciate your reply. Your videos are informative and contain a lot of good information. I've been trying to write a custom handler for an Android specific control (EditText) and I'm having trouble consuming it in my XAML. I've included my namespace at the type of my xaml file. xmlns:mynamespace and then I've been trying to use the control in my XAML like "mynamespace:CustomTextViewHandler". I've been receiving a "System.InvalidOperationException: 'A suitable constructor for type 'TestSipSorcery.CustomTextViewHandler' could not be located". Any advice would be greatly appreciated

**Reply by pictos** [reply]
Hey @headintheclouds21 can you share this project on github? That way I can take a look
